crates/assists/src/handlers/generate_getter.rs

```diff
@@ -0,0 +1,159 @@
+use ast::Adt;
+use stdx::{format_to, to_lower_snake_case};
+use syntax::ast::VisibilityOwner;
+use syntax::ast::{self, AstNode, NameOwner};
+
+use crate::{
+    utils::{find_impl_block, find_struct_impl, generate_impl_text},
+    AssistContext, AssistId, AssistKind, Assists,
+};
+
+// Assist: generate_getter
+//
+// Generate a getter method.
+//
+// ```
+// struct Person {
+//     nam$0e: String,
+// }
+// ```
+// ->
+// ```
+// struct Person {
+//     name: String,$0
+// }
+//
+// impl Person {
+//     /// Get a reference to the person's name.
+//     fn name(&self) -> &String {
+//         &self.name
+//     }
+// }
+// ```
+pub(crate) fn generate_getter(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
+    let strukt = ctx.find_node_at_offset::<ast::Struct>()?;
+    let field = ctx.find_node_at_offset::<ast::RecordField>()?;
+
+    let strukt_name = strukt.name()?;
+    let field_name = field.name()?;
+    let field_ty = field.ty()?;
+
+    let fn_name = to_lower_snake_case(&field_name.to_string());
+    let fn_name_spaced = fn_name.replace('_', " ");
+    let strukt_name_spaced = to_lower_snake_case(&strukt_name.to_string()).replace('_', " ");
```

Comment:
This can be put into the closure, no need to construct these strings if the assist doesn't run. Same applies for the other assists.

Reply:
ah yeah good call; thanks!